planner/core/logical_plan_builder.go

```diff
@@ -2290,6 +2297,26 @@ func (b *PlanBuilder) popTableHints() {
 	b.tableHintInfo = b.tableHintInfo[:len(b.tableHintInfo)-1]
 }
 
+func (b *PlanBuilder) appendUnmatchedIndexHintWarning(indexHints []indexHintInfo, usedForIndexMerge bool) {
+	for _, hint := range indexHints {
+		if !hint.matched {
+			var hintTypeString string
+			if usedForIndexMerge {
+				hintTypeString = "index_merge"
+			} else {
+				hintTypeString = hint.hintTypeString()
+			}
+			errMsg := fmt.Sprintf("%s(%s) is inapplicable, check whether the table(%s.%s) exists",
```

Comment:
After thinking again, how about only produce one warning for all the inapplicable index hints? It would help to reduce the warning count when a lot of hints are inapplicable.

Reply:
IMHO, that's not really necessary. One warning for one hint is more helpful to identify the syntax error.